### ISSUE TITLE
Missing bijectors?

### ISSUE THREAD

**Issue by fleischd** [tensorflow/probability#768]
I'm not seeing a bunch of bisectors; am I missing something?

import tensorflow_probability as tfp  
tfd = tfp.distributions  
tfb = tfp.bijectors  
dir(tfb)

yields:

['AbsoluteValue',
 'Affine',
 'AffineLinearOperator',
 'AffineScalar',
 'AutoregressiveLayer',
 'AutoregressiveNetwork',
 'BatchNormalization',
 'Bijector',
 'Blockwise',
 'Chain',
 'CholeskyOuterProduct',
 'CholeskyToInvCholesky',
 'CorrelationCholesky',
 'Cumsum',
 'DiscreteCosineTransform',
 'Exp',
 'Expm1',
 'FillTriangular',
 'Gumbel',
 'Identity',
 'Inline',
 'Invert',
 'IteratedSigmoidCentered',
 'Kumaraswamy',
 'MaskedAutoregressiveFlow',
 'MatrixInverseTriL',
 'MatvecLU',
 'NormalCDF',
 'Ordered',
 'Permute',
 'PowerTransform',
 'RationalQuadraticSpline',
 'RealNVP',
 'Reciprocal',
 'Reshape',
 'ScaleTriL',
 'Sigmoid',
 'SinhArcsinh',
 'Softfloor',
 'SoftmaxCentered',
 'Softplus',
 'Softsign',
 'Square',
 'Tanh',
 'TransformDiagonal',
 'Transpose',
 'Weibull',
 '__all__',
 '__builtins__',
 '__cached__',
 '__doc__',
 '__file__',
 '__loader__',
 '__name__',
 '__package__',
 '__path__',
 '__spec__',
 'absolute_import',
 'absolute_value',
 'affine',
 'affine_linear_operator',
 'affine_scalar',
 'batch_normalization',
 'bijector',
 'blockwise',
 'chain',
 'cholesky_outer_product',
 'cholesky_to_inv_cholesky',
 'correlation_cholesky',
 'cumsum',
 'discrete_cosine_transform',
 'division',
 'exp',
 'expm1',
 'fill_triangular',
 'gumbel',
 'identity',
 'inline',
 'invert',
 'iterated_sigmoid_centered',
 'kumaraswamy',
 'masked_autoregressive',
 'masked_autoregressive_default_template',
 'masked_dense',
 'matrix_inverse_tril',
 'matveclu',
 'normal_cdf',
 'ordered',
 'permute',
 'power_transform',
 'print_function',
 'rational_quadratic_spline',
 'real_nvp',
 'real_nvp_default_template',
 'reciprocal',
 'reshape',
 'scale_tril',
 'sigmoid',
 'sinh_arcsinh',
 'softfloor',
 'softmax_centered',
 'softplus',
 'softsign',
 'square',
 'tanh',
 'transform_diagonal',
 'transpose',
 'weibull']

Seems I'm missing "Scale()", among others. Or are these deprecated now?

**Comment by csuter** [comment]
what's your `pip list` output?

**Comment by fleischd** [comment]
# Name                    Version                   Build  Channel
_tflow_select             2.3.0                       mkl  
absl-py                   0.8.1                    py37_0  
appnope                   0.1.0                    py37_0  
asn1crypto                1.3.0                    py37_0  
astor                     0.8.0                    py37_0  
astroid                   2.3.3                    py37_0  
attrs                     19.3.0                     py_0  
backcall                  0.1.0                    py37_0  
beautifulsoup4            4.8.2                    py37_0  
blas                      1.0                         mkl  
bleach                    3.1.0                    py37_0  
c-ares                    1.15.0            h1de35cc_1001  
ca-certificates           2020.1.1                      0  
cachetools                4.0.0                    pypi_0    pypi
certifi                   2019.11.28               py37_0  
cffi                      1.13.2           py37hb5b8e2f_0  
chardet                   3.0.4                 py37_1003  
cloudpickle               1.2.2                    pypi_0    pypi
conda                     4.8.2                    py37_0  
conda-package-handling    1.6.0            py37h1de35cc_0  
cryptography              2.8              py37ha12b0ac_0  
cycler                    0.10.0                   py37_0  
dbus                      1.13.12              h90a0687_0  
decorator                 4.4.1                      py_0  
defusedxml                0.6.0                      py_0  
dm-sonnet                 1.23                     pypi_0    pypi
entrypoints               0.3                      py37_0  
expat                     2.2.6                h0a44026_0  
freetype                  2.9.1                hb4e5f40_0  
gast                      0.2.2                    py37_0  
gettext                   0.19.8.1             h15daf44_3  
glib                      2.63.1               hd977a24_0  
google-auth               1.11.0                   pypi_0    pypi
google-auth-oauthlib      0.4.1                    pypi_0    pypi
google-pasta              0.1.8                      py_0  
grpcio                    1.16.1           py37h044775b_1  
h5py                      2.10.0           py37h3134771_0  
hdf5                      1.10.4               hfa1e0ec_0  
icu                       58.2                 h4b95b61_1  
idna                      2.8                      py37_0  
importlib_metadata        1.5.0                    py37_0  
inflect                   4.1.0                    py37_0  
intel-openmp              2019.4                      233  
ipykernel                 5.1.4            py37h39e3cac_0  
ipython                   7.12.0           py37h5ca1d4c_0  
ipython_genutils          0.2.0                    py37_0  
ipywidgets                7.5.1                      py_0  
isort                     4.3.21                   py37_0  
jaraco.itertools          5.0.0                      py_0  
jedi                      0.16.0                   py37_0  
jinja2                    2.11.1                     py_0  
jpeg                      9b                   he5867d9_2  
jsonschema                3.2.0                    py37_0  
jupyter                   1.0.0                    py37_7  
jupyter_client            5.3.4                    py37_0  
jupyter_console           6.1.0                      py_0  
jupyter_core              4.6.1                    py37_0  
keras-applications        1.0.8                      py_0  
keras-preprocessing       1.1.0                      py_1  
kiwisolver                1.1.0            py37h0a44026_0  
lazy-object-proxy         1.4.3            py37h1de35cc_0  
libcxx                    4.0.1                hcfea43d_1  
libcxxabi                 4.0.1                hcfea43d_1  
libedit                   3.1.20181209         hb402a30_0  
libffi                    3.2.1                h475c297_4  
libgfortran               3.0.1                h93005f0_2  
libiconv                  1.15                 hdd342a3_7  
libpng                    1.6.37               ha441bb4_0  
libprotobuf               3.11.2               hd9629dc_0  
libsodium                 1.0.16               h3efe00b_0  
libxml2                   2.9.9                hf6e021a_1  
libxslt                   1.1.33               h33a18ac_0  
lxml                      4.4.2            py37hef8c89e_0  
markdown                  3.1.1                    py37_0  
markupsafe                1.1.1            py37h1de35cc_0  
matplotlib                3.1.1            py37h54f8f79_0  
mccabe                    0.6.1                    py37_1  
mistune                   0.8.4            py37h1de35cc_0  
mkl                       2019.4                      233  
mkl-service               2.3.0            py37hfbe908c_0  
mkl_fft                   1.0.15           py37h5e564d8_0  
mkl_random                1.1.0            py37ha771720_0  
more-itertools            8.2.0                      py_0  
multitasking              0.0.9                      py_0    ranaroussi
nbconvert                 5.6.1                    py37_0  
nbformat                  5.0.4                      py_0  
ncurses                   6.1                  h0a44026_1  
notebook                  6.0.3                    py37_0  
numpy                     1.18.1           py37h7241aed_0  
numpy-base                1.18.1           py37h6575580_1  
oauthlib                  3.1.0                    pypi_0    pypi
openssl                   1.1.1d               h1de35cc_3  
opt_einsum                3.1.0                      py_0  
pandas                    0.25.3           py37h0a44026_0  
pandoc                    2.2.3.2                       0  
pandocfilters             1.4.2                    py37_1  
parso                     0.6.0                      py_0  
pcre                      8.43                 h0a44026_0  
pexpect                   4.8.0                    py37_0  
pickleshare               0.7.5                    py37_0  
pip                       20.0.2                   py37_1  
prometheus_client         0.7.1                      py_0  
prompt_toolkit            3.0.3                      py_0  
protobuf                  3.11.2           py37h0a44026_0  
ptyprocess                0.6.0                    py37_0  
pyarrow                   0.15.1                   pypi_0    pypi
pyasn1                    0.4.8                    pypi_0    pypi
pyasn1-modules            0.2.8                    pypi_0    pypi
pycosat                   0.6.3            py37h1de35cc_0  
pycparser                 2.19                     py37_0  
pygments                  2.5.2                      py_0  
pylint                    2.4.4                    py37_0  
pyopenssl                 19.1.0                   py37_0  
pyparsing                 2.4.6                      py_0  
pyqt                      5.9.2            py37h655552a_2  
pyrsistent                0.15.7           py37h1de35cc_0  
pysocks                   1.7.1                    py37_0  
python                    3.7.4                h359304d_1  
python-dateutil           2.8.1                      py_0  
python.app                2                        py37_9  
pytz                      2019.3                     py_0  
pyzmq                     18.1.0           py37h0a44026_0  
qt                        5.9.7                h468cd18_1  
qtconsole                 4.6.0                      py_1  
readline                  7.0                  h1de35cc_5  
requests                  2.22.0                   py37_1  
requests-oauthlib         1.3.0                    pypi_0    pypi
rsa                       4.0                      pypi_0    pypi
ruamel_yaml               0.15.87          py37h1de35cc_0  
scipy                     1.4.1                    pypi_0    pypi
send2trash                1.5.0                    py37_0  
setuptools                45.1.0                   py37_0  
simfin                    0.4.0                    pypi_0    pypi
sip                       4.19.8           py37h0a44026_0  
six                       1.14.0                   py37_0  
soupsieve                 1.9.5                    py37_0  
sqlite                    3.31.1               ha441bb4_0  
tensorboard               2.1.0                    pypi_0    pypi
tensorflow                2.1.0                    pypi_0    pypi
tensorflow-base           2.0.0           mkl_py37h66b1bf0_0  
tensorflow-estimator      2.1.0                    pypi_0    pypi
tensorflow-probability    0.9.0                    pypi_0    pypi
termcolor                 1.1.0                    py37_1  
terminado                 0.8.3                    py37_0  
testpath                  0.4.4                      py_0  
tk                        8.6.8                ha441bb4_0  
tornado                   6.0.3            py37h1de35cc_0  
tqdm                      4.42.0                     py_0  
traitlets                 4.3.3                    py37_0  
urllib3                   1.25.8                   py37_0  
wcwidth                   0.1.8                      py_0  
webencodings              0.5.1                    py37_1  
werkzeug                  0.16.1                     py_0  
wheel                     0.34.2                   py37_0  
widgetsnbextension        3.5.1                    py37_0  
wrapt                     1.11.2           py37h1de35cc_0  
xz                        5.2.4                h1de35cc_4  
yaml                      0.1.7                hc338f04_2  
yfinance                  0.1.54                     py_0    ranaroussi
zeromq                    4.3.1                h0a44026_3  
zipp                      2.1.0                      py_0  
zlib                      1.2.11               h1de35cc_3

**Comment by csuter** [comment]
Weird. I pip installed tensorflow-probability 0.9 in a fresh virtualenv and I have a longer list of Bijectors, including Scale and its various friends. Can you try uninstalling and reinstalling?

**Comment by davmre** [comment]
Second Chris that the most likely explanation is that you're somehow importing an old version of TFP instead of the new one that pip knows about. You can check this by `print(tfp.__version__)`. Assuming that's the problem, doing a fresh install in a clean virtualenv should fix things.

**Comment by fleischd** [comment]
Yes, just tried that. All fixed. Very strange. Thanks.
